### PR TITLE
Add/new performance indicators

### PR DESCRIPTION
**Pull request by sambua**
## Proposed changes
- Added a new performance analytics for the player states

## Checklist
- [x] Tested locally
- [x] Added new headless integration tests
- [ ] Updated docs

### CODE REVIEW

File: src/dynamics/ads_player/ads_player.js [hunk 1]
```diff
@@ -116,7 +116,7 @@ Scoped.define("module:Ads.Dynamics.Player", [
                         const dyn = this.parent();
                         if (dyn) {
                             dyn.stopAdsRenderFailTimeout(true);
-                            dyn.channel("ads").trigger("ad-error");
+                            dyn.trigger("ad-error", `Render ads took a too much time`);
```
Comment (ntshcalleia) [inline src/dynamics/ads_player/ads_player.js:119]: @sambua shouldn't we be sending ad data with this event?
```suggestion
                            dyn.trigger("ad-error", "Ad took too long to render");
```

File: src/dynamics/video_player/player/player.js [hunk 2]
```diff
@@ -3296,7 +3334,53 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
...
+                                if (entry[key]) perf[key] = entry[key];
+                            });
+                        });
+                        // to be able to trigger change event
+                        if (index > -1) this.get(`performancerecords`)[index] = perf;
+                        this.set(`performancerecords`, index > -1 ? this.get(`performancerecords`) : [
+                            ...this.get(`performancerecords`),
+                            perf
+                        ]);
+                    }, this);
```
Comment (ntshcalleia) [inline src/dynamics/video_player/player/player.js:3383]: Hi @sambua, we should be able to unit test this

**Review by ntshcalleia — CHANGES_REQUESTED** [review]

File: src/dynamics/video_player/player/player.js (hunk 2 above)
Comment (sambua) [reply src/dynamics/video_player/player/player.js:3383]: Hi @ntshcalleia as js-dom not fully supporting performance API added headless integration test

**Review by ntshcalleia — APPROVED** [review]

File: package-source.json [hunk 3]
```diff
@@ -63,6 +63,7 @@
         "test": "npm run test:traced && npm run test:show-report",
         "test:unit": "nyc qunit --require ./tests/unit/index.js ./tests/unit/**/**.spec.js",
         "test:ui": "npx playwright test --ui --config ./tests/playwright/config.js",
+        "test:unit": "nyc qunit --require ./tests/unit/index.js ./tests/unit/**/**.spec.js",
```
[the remark this answers is not in the document]
Comment (sambua) [reply package-source.json:66]: Hi @ntshcalleia , thanks! Seems it was after rebasing with master.

**Pull request merged**